crates/dbsp/src/operator/time_series/watermark.rs

```diff
@@ -120,4 +221,51 @@ mod tests {
     fn test_watermark_monotonic4() {
         test_watermark_monotonic(4);
     }
+
+    fn test_watermark(workers: usize) {
+        let mut expected_watermarks = vec![(-10, 1), (100, 3), (100, 7), (250, 7)].into_iter();
+
+        let (mut dbsp, input_handle) = Runtime::init_circuit(workers, move |circuit| {
+            let (stream, handle) = circuit.add_input_indexed_zset::<i32, i32, _>();
+            stream
+                .watermark(
+                    || (i32::MIN, i32::MIN),
```

Comment:
yes, for some types like dates there may be no obvious min value.

Reply:
why not?  I think date types usually come with the smallest and largest representable dates.